Core/Core/Router/Router.swift

```diff
@@ -89,32 +86,33 @@ public extension RouterProtocol {
 
         if let displayModeButton = from.displayModeButtonItem,
             from.splitViewController?.isCollapsed == false,
-            options.detail || from.isInSplitViewDetail,
-            options.modal == nil {
+            options.isDetail || from.isInSplitViewDetail,
+            !options.isModal {
             view.addNavigationButton(displayModeButton, side: .left)
             view.navigationItem.leftItemsSupplementBackButton = true
         }
 
-        if options.addDoneButton {
-            view.addDoneButton(side: .left)
-        }
         var nav: UINavigationController?
         if options.embedInNav {
             nav = view as? UINavigationController ?? UINavigationController(rootViewController: view)
         }
 
-        if let modal = options.modal {
+        switch options {
+        case let .modal(modalOptions):
```

Comment:
I thought the syntax was `case .modal(let style, let isDismissable, let embedInNav, let addDoneButton)`. Does what you did here just collect all the params into a tuple?

Reply:
It seems to, I didn't want to rely on the order of the variables, it would be easy to accidentally switch a `let isDismissable` with a `let embedInNav` with no warning in the above version.